pkg/cluster/loadbalancerprofile.go

```diff
@@ -263,8 +317,31 @@ func (m *manager) reconcileDesiredManagedIPs(ctx context.Context) ([]api.Resourc
 			break
 		}
 	}
+	return desiredIPAddresses
+}
 
-	return desiredIPAddresses, nil
+func (m *manager) createPublicIPAddresses(ctx context.Context, ipAddresses map[string]mgmtnetwork.PublicIPAddress, numToCreate int) error {
+	ch := make(chan createIPResult)
+	defer close(ch)
+	var errResults []string
+	// create additional IPs if needed
+	for i := 0; i < numToCreate; i++ {
+		go m.createPublicIPAddress(ctx, ch)
+	}
+
+	for i := 0; i < numToCreate; i++ {
+		result := <-ch
+		if result.err != nil {
+			errResults = append(errResults, fmt.Sprintf("creation of ip address %s failed with error: %s", *result.ip.Name, result.err.Error()))
+		} else {
+			ipAddresses[*result.ip.Name] = result.ip
+		}
+	}
+
+	if len(errResults) > 0 {
```

Comment:
I think we can check directly if errResult != nil, a bit more idiomatic and shorter 🙂